data_model/primitives/Cargo.toml

```diff
@@ -18,21 +18,20 @@ maintenance = { status = "actively-developed" }
 
 [features]
 default = ["std"]
-std = ["iroha_macro/std", "iroha_version/std", "iroha_version/warp", "iroha_crypto/std", "fixnum/std", "thiserror"]
-roles = []
+std = ["iroha_macro/std", "fixnum/std", "thiserror"]
 
 [dependencies]
-iroha_crypto = { path = "../../crypto", version = "=2.0.0-pre.1", default-features = false }
 iroha_macro = { path = "../../macro", version = "=2.0.0-pre.1", default-features = false }
 iroha_schema = { path = "../../schema", version = "=2.0.0-pre.1", default-features = false }
-iroha_version = { path = "../../version", version = "=2.0.0-pre.1", default-features = false, features = ["derive", "json", "scale"] }
 
 parity-scale-codec = { version = "2.3.1", default-features = false, features = ["derive"] }
 fixnum = { version = "0.6.1", default-features = false, features = ["serde", "parity", "i64"]}
 derive_more = { version = "0.99.16", default-features = false, features = ["display"] }
 serde = { version = "1.0", default-features = false, features = ["derive"] }
-serde_json = { version = "1.0.59", default-features = false }
-warp = { version = "0.3", default-features = false, optional = true }
+smallvec = { version = "1.8.0", default-features = false, features = ["serde", "union"] }
+# TODO: use specific version (> 0.2.0) once it's available
+smallstr = { git = "https://github.com/murarth/smallstr", rev = "68efc08", default-features = false, features = ["serde", "union"] }
```

Comment:
We have to reference a specific commit, to be able to use `no_std`

Reply:
Makes sense.